### PR TITLE
Add a POC for lib composer inside the main repo for alpha phase

### PR DESCRIPTION
**Pull request by dmanners**
This is a proof of concept for building all libs inside the main repo to start with but with composer to easily extract at a later date.

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -15,14 +16,16 @@
...
+        "symfony/console": "^3.2",
+        "kickasscommerce/authentication": "*@dev"
     },
     "require-dev": {
         "phpunit/phpunit": "*"
     },
     "autoload": {
         "psr-4": {
             "KickAss\\Commerce\\": "src/",
+            "KickAss\\Authentication\\": "src/Authentication",
```
Comment (dmanners) [inline composer.json]: I could not get it to pick the auto loader from the sup repo but maybe you know the magic!

File: src/Authentication/AuthenticatorInterface.php [hunk 2]
```diff
@@ -1,6 +1,6 @@
 <?php
 
-namespace KickAss\Commerce\Application;
+namespace KickAss\Authentication;
```
Comment (dmanners) [inline src/Authentication/AuthenticatorInterface.php]: Folder structure is up for discussion

**Comment by dmanners** [comment]
All thoughts, questions and complaints are welcome. I was also thinking of move the moltin repo back in for the alpha. I know I know that sounds bad.

File: composer.json [hunk 3]
```diff
@@ -4,7 +4,8 @@
     "type": "project",
     "license": "MIT",
     "repositories": [
-        { "type": "vcs", "url": "git@github.com:KickAssCommerce/moltin.git" }
+        { "type": "vcs", "url": "git@github.com:KickAssCommerce/moltin.git" },
+        { "type": "path", "url": "src/*" }
```
Comment (dmanners) [inline composer.json]: could also consider moving these modules to lib

File: composer.json (hunk 1 above)
Comment (sandermangel) [reply composer.json]: @dmanners not that good at composer I'm afraid

Comment (dmanners) [reply composer.json]: I think I remember we had this in a project that we needed to do something like this until we extracted it but I cannot remember.

**Comment by dmanners** [comment]
Will think about more than just Auth and update this pr sometime soon ish maybe

**Comment by dmanners** [comment]
So this now has all current "sub-module" in the main repo. I know we did not want a monolith but for the current development process I would suggest building each module inside src with it's own readme, composer.json etc and then as a final set we extract them all. What do you think @sandermangel and @adam-paterson

File: composer.json [hunk 4]
```diff
@@ -13,17 +13,20 @@
...
+        "kickasscommerce/authentication": "*@dev"
     },
     "require-dev": {
         "phpunit/phpunit": "*",
         "squizlabs/php_codesniffer": "^2.8"
     },
     "autoload": {
         "psr-4": {
-            "KickAss\\Commerce\\": "src/",
+            "KickAss\\Commerce\\": "src/Commerce",
```
Comment (dmanners) [inline composer.json:27]: This is the "main" app

File: composer.json [hunk 5]
```diff
@@ -13,17 +13,20 @@
...
+            "KickAss\\Authentication\\": "src/Authentication/src",
```
Comment (dmanners) [inline composer.json:28]: This is something that we would like to extract later on

**Review by sandermangel — APPROVED** [review]

**Comment by adam-paterson** [comment]
@dmanners I'm fine with this. The other option is to create the repo's for the bundles as we go along and then in your global `composer.json` specify a local path as a repository location.

For multi-package projects this is what I do, and then disable the global composer when I want to test a "proper" build.

**Review by adam-paterson — APPROVED** [review]
@dmanners This is fine, the other option is to create the bundle repo's as we go along and then in your global `composer.json` override the `repositories` with a local path.

This is what I do for multi-package projects at the moment.

**Pull request merged**